test/cpp/util/test_credentials_provider.h

```diff
@@ -68,6 +74,10 @@ class CredentialsProvider {
 
   // Provide a list of secure credentials type.
   virtual std::vector<std::string> GetSecureCredentialsTypeList() = 0;
+
+  // Provide a transport security type given a credentials type.
+  virtual std::string GetTransportSecurityType(
```

Comment:
I think this function can be removed from the class and made a standalone function (since it does not depend on the partial `CredentialProvider` implementation.)